Explorer/Assets/DCL/SDKComponents/MediaStream/Systems/CreateMediaPlayerSystem.cs

```diff
@@ -44,48 +51,52 @@ protected override void Update(float t)
         [None(typeof(MediaPlayerComponent))]
         private void CreateAudioStream(in Entity entity, ref PBAudioStream sdkComponent)
         {
-            if (!frameTimeBudget.TrySpendBudget()) return;
-
-            MediaPlayerComponent component = CreateMediaPlayerComponent(sdkComponent.Url, sdkComponent.HasVolume, sdkComponent.Volume, autoPlay: sdkComponent.HasPlaying && sdkComponent.Playing);
-            World.Add(entity, component);
+            CreateMediaPlayer(entity, sdkComponent.Url, sdkComponent.HasVolume, sdkComponent.Volume, autoPlay: sdkComponent.HasPlaying && sdkComponent.Playing);
         }
 
         [Query]
         [None(typeof(MediaPlayerComponent))]
         [All(typeof(VideoTextureComponent))]
-        private void CreateVideoPlayer(in Entity entity, ref PBVideoPlayer sdkComponent)
+        private void CreateVideoPlayer(in Entity entity, PBVideoPlayer sdkComponent)
+        {
+            CreateMediaPlayer(entity, sdkComponent.Src, sdkComponent.HasVolume, sdkComponent.Volume, sdkComponent.HasPlaying && sdkComponent.Playing, OnComplete);
+            return;
+
+            void OnComplete(MediaPlayer mediaPlayer) => mediaPlayer.SetPlaybackProperties(sdkComponent);
```

Comment:
It's dangerous: the callback can be called on the component that does not belong to the current context anymore (was pooled) that may lead to random failures.
Any ideas how to handle it without a capture?